(maint) Fix broken acceptance test

justinstoller opened the pull request:
Previously, the single_env_purge_unmanaged_modules test said that
it was testing that agent runs would fail because the motd module
had been successfully purged.

However, it was actually testing that a) the purge command failed and
b) the agent run succeeded but the assertion failed, not because of
anything that happened on SUT, but because it was testing the output
against the regex /Blah/ which should not appear in normal agent output.
The test then claimed that the failure was part of CODEMGMT-78, which
was a feature improvement to `r10k puppetfile` to respect `--moduledir`
and `--puppetfile` flags. However, that feature was implemented in 2016
and the test has continued to "pass" because of a failing purge command
because the test never actually attempted to use either the moduledir
or puppetfile flags.

This patch updates the purge invocation to actually use moduledir and
puppetfile flags, updates the expectation on the purge invocation to
be successful. It also updates the expectation on agent runs to be
unsuccessful but removes the duplicitous expectation on output matching
/Blah/.

Please add all notable changes to the "Unreleased" section of the CHANGELOG in the format:
```
- (JIRA ticket) Summary of changes. [Issue or PR #](link to issue or PR)
```

justinstoller committed:
(maint) Fix broken acceptance test

Previously, the single_env_purge_unmanaged_modules test said that
it was testing that agent runs would fail because the motd module
had been successfully purged.

However, it was actually testing that a) the purge command failed and
b) the agent run succeeded but the assertion failed, not because of
anything that happened on SUT, but because it was testing the output
against the regex /Blah/ which should not appear in normal agent output.
The test then claimed that the failure was part of CODEMGMT-78, which
was a feature improvement to `r10k puppetfile` to respect `--moduledir`
and `--puppetfile` flags. However, that feature was implemented in 2016
and the test has continued to "pass" because of a failing purge command
because the test never actually attempted to use either the moduledir
or puppetfile flags.

This patch updates the purge invocation to actually use moduledir and
puppetfile flags, updates the expectation on the purge invocation to
be successful. It also updates the expectation on agent runs to be
unsuccessful but removes the duplicitous expectation on output matching
/Blah/.

justinstoller committed:
(maint) Update to JRuby 9.2.10.0 for new forge ciphers

With the latest forge updates they only support modern ciphers.
Unfortunately JRuby does not support those ciphers until 9.2.10.

There was some discussion around removing JRuby from the matrix and
while we don't use R10K on JRuby for any purposes internally it seems
like dropping support for a form of Ruby would at least be a major
version boundary decision.

mwaggett commented on integration/tests/user_scenario/basic_workflow/single_env_purge_unmanaged_modules.rb:
```diff
@@ -1,10 +1,11 @@
 
 #Init
 master_certname = on(master, puppet('config', 'print', 'certname')).stdout.rstrip
+moduledir = on(master, puppet('config', '--environment', 'production', 'print', 'environmentpath')).stdout.strip + '/production/modules'
```
iirc, you were just calling `config print moduledir` before, right? Why didn't that work?

I'm surprised you need to include the `production/` dir in the path you're appending - I thought the `environmentpath` would already include the `production/` dir?

justinstoller replied:
Good catch.

Before I was calling moduledir, which isn't a Puppet thing. It's an actually an r10k thing and I was confused, and modulepath could return a ':' separated list of directories to search for modules, so I decided against using that.

I then put in `--environment production print environmentpath` thinking I'd get back something that ended in production. However, this is returning something I thought was called an "environmentspath" but it looks like "environmentspath" isn't a thing. So we have to append the environment name to it. In that case I don't believe the `--environment production` is necessary anymore and can be removed. I'll do that.

Working in file-sync has destroyed my recollection of this part of Puppet's settings, and I was being sloppy because this acceptance angered me so much.

mwaggett replied:
haha all good! I've never had an understanding of Puppet's settings, so no worries 😅 
thanks for explaining!

justinstoller committed:
(maint) Update default OS for integration tests

Previously we tested on CentOS 6 which is no longer a supported platform
on our internal tooling.

This patch continues to use CentOS since it is the traditional platform
for our FOSS project, though due to some concerns about the direction of
CentOS 8 we stay with CentOS 7. When we must move off of it, there will
hopefully be more guidance as to what Enterprise Linux variant to use in
FOSS testing.
